### PR TITLE
correct some configurations for lbaas

### PR DESCRIPTION
**Pull request by panxia6679**
1) use nobody instead of nogroup for rhel
2) correct lbaas symlinks
3) create lbaas service for rhel + distro

### CODE REVIEW

**Comment by nirajdp76** [comment]
ok to test

**Comment by panxia6679** [comment]
Just corrected a type. Would you please retest? Thanks.

File: roles/neutron-common/meta/main.yml [hunk 1]
```diff
@@ -27,3 +27,10 @@ dependencies:
     dependent_packages: "{{ neutron.osp_sys_packages }}"
     rootwrap: True
     when: openstack_install_method == 'distro'
+  - role: openstack-distro
+    project_name: neutron-lbaas
+    dependent_packages: "{{ neutron.osp_sys_packages }}"
+    rootwrap: True
```
Comment (scoop206) [inline roles/neutron-common/meta/main.yml]: I don't think we actually need rootwrap on openstack-distro.  Looks like we us it in a few places though so we can probably leave here and come back and cleanup if needs be.

**Comment by scoop206** [comment]
test this please

**Comment by nirajdp76** [comment]
retest

**Comment by nirajdp76** [comment]
retest

**Comment by nirajdp76** [comment]
retest

**Comment by nirajdp76** [comment]
retest

File: roles/neutron-data-network/tasks/main.yml [hunk 2]
```diff
@@ -262,6 +278,24 @@
         - /etc/neutron/services/loadbalancer/haproxy/lbaas_agent.ini
         - /etc/neutron/neutron_lbaas.conf
 
+- name: install neutron lbaas service (rhel distro)
+  systemd_service:
```
Comment (paulczar) [inline roles/neutron-data-network/tasks/main.yml]: are you sure neutronlbaas package doesn't provide a systemd unit already?  most of the rhel packages do.

File: roles/neutron-common/meta/main.yml (hunk 1 above)
Comment (panxia6679) [inline roles/neutron-common/meta/main.yml]: yep, I agree. Acutally I just refer to others and we can double verify it later and clean up if needs be.

File: roles/neutron-data-network/tasks/main.yml (hunk 2 above)
Comment (panxia6679) [inline roles/neutron-data-network/tasks/main.yml]: yes, package provided a systemd service unit, but the content of the service has some difference to ours. Taking neutron lbaas as example, we copied lbaas_agent.ini to /etc/neutron/services/loadbalancer/haproxy/, however rpm uses location of /etc/neutron/. We have to overwrite these configuration of the service unit. In addtion, service unit creatd by red hat contains -dist.conf files and I removed them by overwriting the service unit.

Comment (panxia6679) [inline roles/neutron-data-network/tasks/main.yml]: yes, package provided a systemd service unit, but the content of the service has some difference to ours. Taking neutron lbaas as example, we copied lbaas_agent.ini to /etc/neutron/services/loadbalancer/haproxy/, however rpm uses location of /etc/neutron/. We have to overwrite these configuration of the service unit. In addtion, service unit creatd by red hat contains -dist.conf files and I removed them by overwriting the service unit.

Comment (panxia6679) [inline roles/neutron-data-network/tasks/main.yml]: answered question above.

File: roles/neutron-common/meta/main.yml (hunk 1 above)
Comment (zfeldstein) [inline roles/neutron-common/meta/main.yml]: I think we could just put the neutron lbaas package into the osp_sys_packages variable in the neutron-common defaults: (https://github.com/blueboxgroup/ursula/blob/centos/roles/neutron-common/defaults/main.yml#L12)

We can install the package no matter if its' enabled or not, and then configure neutron accordingly if we have neutron.lbaas == enabled.

File: roles/neutron-data-network/tasks/main.yml (hunk 2 above)
Comment (zfeldstein) [inline roles/neutron-data-network/tasks/main.yml]: We can probably change the dest path on where we drop these templates to where the OSP packages expect. This is what we had to do for horizon as well. IMO I think it's be better if we could just use the OSP packages as is w/out having to customize the systemd stuff.

**Comment by panxia6679** [comment]
close one as I opened new PR #2469

**Pull request closed without merging**